lib/active_record/session_store.rb

```diff
@@ -33,6 +37,60 @@ def create_table!
         end
         connection.add_index table_name, session_id_column, :unique => true
       end
+
+      def determine_serializer
+        self.serializer ||= :marshal
```

Comment:
I like yours better—it keeps the default behavior the same as legacy (which is what I was going for) but without side affects.  Good call.